src/org/ods/usecase/LeVADocumentUseCase.groovy

```diff
@@ -1264,4 +1272,14 @@ class LeVADocumentUseCase extends DocGenUseCase {
         
         return documentType
     }
+
+    private String getWatermarkText(String documentType) {
+        def environment = this.util.getBuildParams().targetEnvironmentToken
+
+        if (!LeVADocumentScheduler.ENVIRONMENT_TYPE[environment].contains(documentType)){
```

Comment:
The entire watermark feature is really only about the Developer Preview watermark (from the perspective of this class). This watermark only makes sense when we are in the DEV/D environment. Constraining the condition to this environment would be a good idea. It would also allow you to remove the tests for QA and PROD. Further, I would be more specific about the `WATERMARK_MESSAGE` and turn it into a `DEVELOPER_PREVIEW_WATERMARK`.

Reply:
Constrained the environment, modified the constant name and removed extra tests